src/config/config.h

```diff
@@ -0,0 +1,214 @@
+    return core::GetTypeName<data_t>();
+}
+
+template<>
+inline std::string ConfigDataGetStringRep(const std::string& data) {
+    // Ignoring escaping for simplicity.
+    return "\"" + data + "\"";
+}
+
+template<class data_t>
+    requires requires(data_t data) { std::to_string(data); }
+std::string ConfigDataGetStringRep(const data_t& data) {
+    return std::to_string(data);
+}
+
+template<class data_t>
+    requires requires(data_t data) { std::string(data.Str()); }
+std::string ConfigDataGetStringRep(const data_t& data) {
+    return data.Str();
+}
+
+template<class data_t>
+    requires requires(data_t data) { std::string(data->Str()); }
+std::string ConfigDataGetStringRep(const data_t& data) {
+    return data->Str();
+}
+
+template<class T1, class T2>
+std::string ConfigDataGetStringRep(const std::pair<T1, T2>& data) {
+    return "[" + ConfigDataGetStringRep(data.first) + ", " + ConfigDataGetStringRep(data.second) + "]";
+}
+
+template<class data_t>
+    // clang-format off
+    requires(!std::same_as<data_t, std::string>) && requires(data_t data) {
+        std::accumulate(
+            data.begin(),
+            data.end(),
+            std::declval<std::string>(),
+            std::declval<std::function<std::string(std::string, const typename data_t::value_type&)>>());
+    }
+// clang-format on
+std::string ConfigDataGetStringRep(const data_t& data) {
+    return std::accumulate(
+               data.begin(),
+               data.end(),
+               std::string(),
+               [](std::string result, const auto& item) {
+                   result += result.empty() ? "[" : ",";
+                   return std::move(result) + ConfigDataGetStringRep(item);
+               }) +
+        "]";
+}
+
+}  // namespace impl
+
+template<class data_t>
+std::shared_ptr<Config<data_t>> ConfigFile::Get(const std::string& config_name) {
+    return Get(config_name, data_t{});
+}
+
+template<class data_t>
+std::shared_ptr<Config<data_t>> ConfigFile::Get(const std::string& config_name, data_t&& default_value) {
+    std::shared_ptr<ConfigBase> new_config_ptr(new Config<data_t>(config_name));
+
+    auto config_base_ptr = RegisterOrGet(config_name, std::move(new_config_ptr));
+    auto config_ptr      = std::dynamic_pointer_cast<Config<data_t>>(config_base_ptr);
+    if (!config_ptr) {
+        LOG(FATAL) << __func__ << ": Config " << config_name << " is registered, but not " << GetTypeName<data_t>();
+    }
+
+    if (!config_ptr->initialized_.load()) {
+        LOG(INFO) << __func__ << ": Cannot find Config " << config_name << " in the config file. "
+                  << "Use the default value " << impl::ConfigDataGetStringRep(default_value) << " instead.";
+        config_ptr->data_ = std::move(default_value);
+    }
+
+    return config_ptr;
+}
+
+template<class data_t>
+void Config<data_t>::InitValue(simdjson::ondemand::value& val) {
+    if constexpr (std::is_integral_v<data_t>) {
+        if constexpr (!std::is_signed_v<data_t>) {
```

Comment:
do you need to handle bool?

Reply:
No, because parsing error will give you an abort.

Reply:
I mean, do you need to accept bool?